code/modules/reagents/chemistry/reagents/impure_reagents/impure_toxin_reagents.dm

```diff
@@ -0,0 +1,68 @@
+//Reagents produced by metabolising/reacting fermichems inoptimally these specifically are for toxins
+//Inverse = Splitting
+//Invert = Whole conversion
+//Failed = End reaction below purity_min
+
+////////////////////TOXINS///////////////////////////
+
+//Lipolicide
+//impure
+/datum/reagent/impurity/ipecacide
+	name = "Ipecacide"
+	description = "An extremely gross substance that induces vomiting. It is produced when Lipolicide reactions are impure."
+	ph = 7
+	liver_damage = 0
+
+/datum/reagent/impurity/ipecacide/on_mob_metabolize(mob/living/carbon/owner)
+	owner.adjust_disgust(100)
+	return ..()
```

Comment:
rapidly buckling and unbuckling someone from a stasis bed will call this multiple times

also, people can and will give other people with microdoses of this to make their game nigh-unplayable for the next few minutes due to chain-vomiting (if you've ever had the badblink mutation, you'll know exactly what I mean)

Reply:
Good point, either I figure this one out or I remove it. Goal was to have a player vomit once (but really badly). After that there is not supposed to be anything else

Reply:
You can use on_mob_add() I believe!

Reply:
That did the trick @Thalpy
I changed to on_mob_add() and also toned down the value from 100 to 60
Now you won't create a carpet of puke like before, it should be within its limits